Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -119,6 +122,33 @@ class TabScrollingController: NSObject {
             }
         }
     }
+
+    func updateMinimumZoom() {
+        guard let scrollView = scrollView else {
+            return
+        }
+        self.isZoomedOut = roundNum(scrollView.zoomScale) == roundNum(scrollView.minimumZoomScale)
+        self.lastZoomedScale = self.isZoomedOut ? 0 : scrollView.zoomScale
+    }
+
+    func setMinimumZoom() {
+        guard let scrollView = scrollView else {
+            return
+        }
+        if self.isZoomedOut && roundNum(scrollView.zoomScale) != roundNum(scrollView.minimumZoomScale) {
+            scrollView.zoomScale = scrollView.minimumZoomScale
+        }
+    }
+
+    func resetZoomState() {
+        self.isZoomedOut = false
+        self.lastZoomedScale = 0
+    }
+
+    private func roundNum(num: CGFloat) -> CGFloat {
+        return round(100000 * num) / 100000
```

Comment:
`roundNum` lets me round a number to a certain precision. `round(0.89999)` returns 1. But I'd like to keep the numbers after the decimal. 

There is actually a mistake here. This is actually keeping 5 numbers after the decimal. I actually only need 2-3. Sometimes the before/after zoomScale will be 0.8974646/0.8967656 which are basically the same thing.